File: client/src/app/injection/service/injection/injection.service.ts
import {Injectable} from "@angular/core";
import {InjectionHttpService} from "./injection-http.service";
import {NotificationService} from "../../../base/service/notification/notification.service";
import {InjectionId} from "../../models/injection/injection-model";
import {Observable} from "rxjs";
import {InjectionQueueHeader} from "../../models/injection/injection-queue-model";
import {Source} from "../../../base/models/source/source";
import {DocumentFormat} from "../../../base/models/document/document-format";

@Injectable()
export class InjectionService {

    constructor(
        private readonly http: InjectionHttpService,
        private readonly notification: NotificationService
    ) {
    }

    public createQueue() {
        return this.http.createQueue();
    }

    public removeQueue(queueId: number) {
        return this.http.removeQueue(queueId);
    }

    public loadHeaders(): Observable<InjectionQueueHeader[]> {
        return this.http.loadHeaders();
    }

    public loadQueue(queueId: number) {
        return this.http.loadQueue(queueId);
    }

    public injectAll(queueId: number) {
        return this.http.injectAll(queueId);
    }

    public createInjections(queueId: number, count: number) {
        return this.http.createInjections(queueId, count);
    }

    public createFromTemplate(queueId: number, file: File) {
        return this.http.createFromTemplate(queueId, file);
    }

    public clearQueue(queueId: number) {
        return this.http.clearQueue(queueId);
    }

    public changeQueueName(queueId: number, name: string) {
        return this.http.changeQueueName(queueId, name);
    }

    public uploadDataFile(injectionId: InjectionId, data: File) {
        return this.http.uploadDataFile(injectionId, data);
    }

    public uploadTemplateFile(injectionId: InjectionId, data: File) {
        return this.http.uploadTemplateFile(injectionId, data);
    }

    public uploadDataSource(injectionId: InjectionId, data: Source) {
        return this.http.uploadDataSource(injectionId, data);
    }

    public uploadTemplateSource(injectionId: InjectionId, data: Source) {
        return this.http.uploadTemplateSource(injectionId, data);
    }

    public inject(injectionId: InjectionId) {
        return this.http.inject(injectionId);
    }

    public validateTemplate(injectionId: number) {
        return this.http.validateTemplate(injectionId);
    }

    public downloadResult(injectionId: InjectionId) {
        return this.http.downloadResult(injectionId);
    }

    public editResultName(injectionId: InjectionId, name: string) {
        return this.http.editResultName(injectionId, name);
    }

    public changeOutputFormat(injectionId: InjectionId, format: DocumentFormat) {
        return this.http.changeOutputFormat(injectionId, format);
    }

    public copy(injectionId: InjectionId) {
        return this.http.copy(injectionId);
    }

    public remove(injectionId: InjectionId) {
        return this.http.remove(injectionId);
    }

    public moveInjection(queueId: number, fromOrderId: number, toOrderId: number) {
        return this.http.moveInjection(queueId, fromOrderId, toOrderId);
    }
}

File: corvus-app/src/main/java/by/varyvoda/corvus/app/service/injection/InjectionService.java
package by.varyvoda.corvus.app.service.injection;

import by.varyvoda.corvus.api.format.DocumentFormat;
import by.varyvoda.corvus.app.model.injection.InjectionQueue;
import by.varyvoda.corvus.app.model.source.Source;
import by.varyvoda.corvus.app.model.user.User;

import java.util.List;
import java.util.function.Consumer;

public interface InjectionService {

    InjectionQueue loadQueue(Integer id, User user);

    List<InjectionQueue> getUserQueues(User user);

    void injectAll(Integer queueId, Consumer<InjectionQueue.Change> progressConsumer, User user);

    InjectionQueue createQueue(User user);

    void removeQueue(Integer queueId, User user);

    InjectionQueue clearQueue(Integer queueId, User user);

    InjectionQueue.Change changeQueueName(Integer queueId, String name, User user);

    InjectionQueue.Change moveInjection(Integer queueId, Integer fromOrderId, Integer toOrderId, User user);

    InjectionQueue.Change createInjections(Integer queueId, int count, User user);

    InjectionQueue.Change setDataSource(Integer injectionId, Source source, User user);

    InjectionQueue.Change setTemplateSource(Integer injectionId, Source source, User user);

    InjectionQueue.Change setPreferredResultName(Integer injectionId, String name, User user);

    InjectionQueue.Change setOutputFormat(Integer injectionId, DocumentFormat format, User user);

    InjectionQueue.Change inject(Integer injectionId, User user);

    InjectionQueue.Change copy(Integer injectionId, User user);

    InjectionQueue.Change remove(Integer injectionId, User user);

    Source getResultSource(Integer injectionId, User user);
}
